lib/webauthn/public_key_credential/creation_options.rb

```diff
@@ -0,0 +1,109 @@
+# frozen_string_literal: true
+
+require "cose/algorithm"
+require "webauthn/public_key_credential/options"
+require "webauthn/public_key_credential/rp_entity"
+require "webauthn/public_key_credential/user_entity"
+
+module WebAuthn
+  # TODO: make keyword arguments mandatory in next major version
+  def self.credential_creation_options(rp_name: nil, user_name: "web-user", display_name: "web-user", user_id: "1")
+    warn(
+      "DEPRECATION WARNING: `WebAuthn.credential_creation_options` is deprecated."\
+      " Please use `WebAuthn::PublicKeyCredential.create_options` instead."
+    )
+
+    PublicKeyCredential::CreationOptions.new(
+      rp: { name: rp_name },
+      user: { id: user_id, name: user_name, display_name: display_name }
+    ).as_json
+  end
+
+  class PublicKeyCredential
+    def self.create_options(options, encoding: :base64url)
+      WebAuthn::PublicKeyCredential::CreationOptions.new(options.merge(encoding: encoding))
```

Comment:
Initially I tried to make this `WebAuthn::PublicKeyCredential::CreationOptions.new(options, encoding: encoding)` so that we divided logically the PK options from the "ruby method options", but I drop it after realizing I was not going to be able to use keyword arguments in the `initialize` method at all.

Open to any other ideas...